.github/workflows/test_qlib_from_source_fast.yml

```diff
@@ -8,42 +8,58 @@ on:
 
 jobs:
   build:
-    timeout-minutes: 120
+
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        os: [windows-latest, ubuntu-18.04, ubuntu-20.04]
+        os: [windows-latest, ubuntu-18.04, ubuntu-20.04, macos-11, macos-latest]
         # not supporting 3.6 due to annotations is not supported https://stackoverflow.com/a/52890129
         python-version: [3.7, 3.8]
 
     steps:
-    - uses: actions/checkout@v2
+    - name: Test qlib from source fast
+      uses: actions/checkout@v2
 
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v2
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Lint with Black
+    - name: Install pytorch according to the system
+      if: ${{ matrix.os == 'macos-11' || matrix.os == 'macos-latest' }}
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install torch torchvision torchaudio
+        brew install libomp  # lightgbm dependencies
+
+    - name: Install pytorch according to the system
+      if: ${{ matrix.os == 'ubuntu-18.04' || matrix.os == 'ubuntu-20.04' }}
+      run: |
+        python -m pip install --upgrade pip
```

Comment:
why `python -m pip install --upgrade pip`  is required now? 
Please give some comments about this

Reply:
Consider here that due to the low version of pip, it may cause problems with the installation package, so upgrade pip before using it.

Reply:
Maybe we can add a section for upgrading pip and give it some comments. 
It looks weird to upgrade pip every time.